packages/cutler/bin/cutler.dart

```diff
@@ -0,0 +1,279 @@
+import 'dart:io';
+
+import 'package:cutler/config.dart';
+import 'package:cutler/model.dart';
+import 'package:path/path.dart' as p;
+
+enum Paths {
+  engineDEPS('DEPS'),
+  flutterEngineVersion('bin/internal/engine.version');
+
+  const Paths(this.path);
+  final String path;
+}
+
+String runCommand(String executable, List<String> arguments,
+    {String? workingDirectory}) {
+  if (config.verbose) {
+    final workingDirectoryString = workingDirectory == null ||
+            p.equals(workingDirectory, Directory.current.path)
+        ? ""
+        : " (in $workingDirectory)";
+    print("$executable ${arguments.join(' ')}$workingDirectoryString");
+  }
+  final result = Process.runSync(executable, arguments,
+      workingDirectory: workingDirectory);
+  if (result.exitCode != 0) {
+    throw Exception("Failed to run $executable $arguments: ${result.stderr}");
+  }
+  return result.stdout.toString().trim();
+}
+
+void dryRunCommand(String executable, List<String> arguments,
+    {String? workingDirectory}) {
+  print("$executable ${arguments.join(' ')}");
+}
+
+extension RepoCommands on Repo {
+  Version versionFrom(String hash) {
+    return Version(hash: hash, repo: this, aliases: getTagsFor(hash));
+  }
+
+  String get _workingDirectory => "${config.checkoutsRoot}/$path";
+
+  String getLatestCommit(String branch) {
+    return runCommand("git", ["log", "-1", "--pretty=%H", branch],
+        workingDirectory: _workingDirectory);
+  }
+
+  List<String> getTagsFor(String commit) {
+    final output = runCommand("git", ["tag", "--points-at", commit],
+        workingDirectory: _workingDirectory);
+    if (output.isEmpty) {
+      return [];
+    }
+    return output.split('\n');
+  }
+
+  Version getForkPoint() {
+    // Could use --fork-point?
```

Comment:
Is this comment needed still? Looks like we are using `--fork-point`?

Reply:
Fixed.